tests/visualization_tests/matplotlib_tests/test_contour.py

```diff
@@ -288,3 +289,91 @@ def test_contour_subplots_have_correct_axis_labels_and_ranges() -> None:
         xlims = [ax.get_xlim() for ax in subplots[:, index]]
         assert all_equal(xlims)
         assert all(range_covers(param_range_with_padding, xlim) for xlim in xlims)
+
+
+@pytest.mark.parametrize("value", [float("inf"), -float("inf")])
+def test_nonfinite_removed(recwarn: WarningsRecorder, value: float) -> None:
+
+    study = create_study()
+    study.add_trial(
+        create_trial(
+            value=0.0,
+            params={"param_a": 1.0, "param_b": 0.0},
+            distributions={
+                "param_a": FloatDistribution(0.0, 1.0),
+                "param_b": FloatDistribution(0.0, 1.0),
+            },
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=1.0,
+            params={"param_a": 0.0, "param_b": 10.0},
+            distributions={
+                "param_a": FloatDistribution(0.0, 1.0),
+                "param_b": FloatDistribution(0.0, 10.0),
+            },
+        )
+    )
+    study.add_trial(
+        create_trial(
+            value=value,
+            params={"param_a": 0.0, "param_b": 1.0},
+            distributions={
+                "param_a": FloatDistribution(0.0, 1.0),
+                "param_b": FloatDistribution(0.0, 1.0),
+            },
+        )
+    )
+
+    # To check if contour lines have been rendered (meaning +-inf trials were removed),
+    # we should be looking if artists responsible for drawing them are preset in the final plot.
+    # Turns out it's difficult to do reliably (No information which artists are responsible for
+    # drawing contours) so instead we are checking for warning raised by matplotlib
+    # when contour plot fails. TODO(xadrianzetx) Find a better way to test this.
```

Comment:
Thank you for your investigation and detailed comments! 
By the way, we can check wheather or not to be used a trial with `inf` value by plot by seeing the scatter plot's data:

```python
from matplotlib.collections import PathCollection
figure.findobj(PathCollection)[-1].get_offsets().data

#       param_a, param_b
# array([[ 1.,  0.],   # trial 0
#        [ 0., 10.]]),    # trial 1
# thanks to this PR, trial 2's param not in here.

# on the master branch, the retuned value is as follows:
# array([[ 1.,  0.], # trial 0
#        [ 0., 10.], # trial 1
#        [ 0.,  1.]])  # trial 2
```

What do you think about this?

Reply:
I was considering this but found that:
* Specific artists that make up a plot can change between minor matplotlib version (see prev commit in this PR that was ok on my local env but failed CI), so targeting them might result in some CI fails down the line.
* `PathCollection` is a generic artist and I found that it might not contain data even if plot renders correctly. I'm also not sure if `findobj` guarantees order in which `PathCollection` instances are returned, so checking specific one might result in some flakiness.

That's why at the moment I feel like testing for warning msg is safer, although I will consider other options later on.

Reply:
hmm, thank you for your explanation.